astropy/coordinates/baseframe.py

```diff
@@ -756,16 +762,25 @@ def realize_frame(self, representation):
         representation : BaseRepresentation
             The representation to use as the data for the new frame.
 
+        copy : bool
+            If True, return a new object describing the same frame with new data.
+            If False, change the data of the current frame instance.
+
         Returns
         -------
         frameobj : same as this frame
-            A new object with the same frame attributes as this one, but
+            A object with the same frame attributes as this one, but
             with the ``representation`` as the data.
         """
-        frattrs = dict([(attr, getattr(self, attr))
-                        for attr in self.get_frame_attr_names()
-                        if attr not in self._attr_names_with_defaults])
-        return self.__class__(representation, **frattrs)
+        if copy:
+            frattrs = dict([(attr, getattr(self, attr))
+                            for attr in self.get_frame_attr_names()
+                            if attr not in self._attr_names_with_defaults])
+            return self.__class__(representation, **frattrs)
+
+        else:
+            self._set_data(representation)
```

Comment:
Do you need `self._rep_cache = dict()` first to clear the cache from any other reprs that might be there?